src/main/java/com/fvv/bookstore/controller/MagazineControllerImpl.java

```diff
@@ -0,0 +1,121 @@
+package com.fvv.bookstore.controller;
+
+import java.util.List;
+
+import com.fvv.bookstore.bean.Magazine;
+import com.fvv.bookstore.dao.MagazineDAO;
+import com.fvv.bookstore.dao.MagazineDAOImpl;
+import com.fvv.bookstore.exception.ControllerException;
+import com.fvv.bookstore.exception.DaoException;
+import com.fvv.bookstore.exception.magazine.MagazineNotFoundException;
+import com.fvv.bookstore.exception.magazine.MagazineValidationException;
+
+/**
+ * Controller class for a Magazine object.
+ * 
+ * @author Fatima Vasquez
+ * <p>Created on 30 de mar de 2018</p>	
+ * @version 1.0 	
+ *
+ */
+public class MagazineControllerImpl implements MagazineController {
+	
+	private MagazineDAO magazineDao;
+	
+	/**
+	 * Class constructor instantiating a new MagazineDAOImpl object.
+	 */
+	public MagazineControllerImpl() {
+		this.magazineDao = new MagazineDAOImpl();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public boolean addMagazine(Magazine magazine) throws 
+		MagazineValidationException, ControllerException {
+		try {
+			this.validateMagazine(magazine);
+			return this.magazineDao.addMagazine(magazine);
+		} catch (DaoException e) {
+			throw new ControllerException("Error to add a magazine", e);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public List<Magazine> listMagazines() throws ControllerException {
+		try {
+			return this.magazineDao.listMagazines();
+		} catch (DaoException e) {
+			throw new ControllerException("Error to load the list", e);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public boolean updateMagazine(Magazine magazine) throws MagazineValidationException, ControllerException {
+		try {
+			this.validateMagazine(magazine);
+			return this.magazineDao.updateMagazine(magazine);
+		} catch (DaoException e) {
+			throw new ControllerException("Error to update a magazine", e);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void removeMagazine(Long id) throws ControllerException {
+		try {
+			this.magazineDao.removeMagazine(id);
+		} catch (DaoException e) {
+			throw new ControllerException("Error to delete a magazine", e);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public Magazine findMagazine(Long id) throws MagazineNotFoundException, ControllerException {
+		try {
+			return this.magazineDao.findMagazine(id);
+		} catch (DaoException e) {
+			throw new ControllerException("Error to find a magazine", e);
+		}
+	}
+	
+	/**
+	 * Validate if a field is empty.
+	 * 
+	 * @param magazine of type Magazine
+	 * @throws MagazineValidationException if a field is empty.
+	 */
+	private void validateMagazine(final Magazine magazine) throws MagazineValidationException {
+		StringBuilder sb = new StringBuilder();
+		if(magazine.getName().equals("")) {
+			sb.append("Field name cannot be empty.").append("\n");			
```

Comment:
Create a utility class called `Constants` and then add the property:

```
private static final String LINE_SEPARATOR = System.lineSeparator();
```

Finally replace all `\n` occurrences in the  code

Reply:
Ok